src/ApiClient.php

```diff
@@ -125,86 +113,24 @@ public function getPermanentIdentifier(string $correlationId): string
     {
         /** @var StatusEndpoint $endpoint */
-        $endpoint = $this->container->get('status');
-        $endpoint->setToken($this->getToken());
+        $endpoint = $this->apiFactory->createEndpoint(StatusEndpoint::class);
 
         return $endpoint->getTranslationRequestStatus($permanentId);
     }
 
+    /**
+     * @inheritDoc
+     */
     public function downloadFile(string $url): string
     {
-        /** @var Download $downloader */
-        $downloader = $this->container->get('file');
-        $downloader->setToken($this->getToken());
+        $downloader = $this->apiFactory->createDownload();
 
         return $downloader->downloadFile($url);
     }
 
-    private function createContainer(
-        ClientInterface $httpClient,
-        RequestFactoryInterface $requestFactory,
-        StreamFactoryInterface $streamFactory,
-    ): void {
-        $container = new Container();
-
-        // Endpoint services are not shared, meaning that a new instance is
-        // created every time the service is requested from the container.
-        // We're doing this because such a service might be called more than
-        // once during the lifetime of a request, so internals set in a previous
-        // usage may leak into the later usages.
-
-        // Add a common REST service to be used by all endpoints.
-        $container->add('rest', Rest::class)
-            ->addArguments([
-                $httpClient,
-                $requestFactory,
-                $streamFactory,
-            ]);
-
-        // Add endpoint services.
-        $endpoints = [
-            'main' => MainEndpoint::class,
-            'referenceData' => ReferenceDataEndpoint::class,
-            'validate' => ValidateEndpoint::class,
-            'requests' => RequestsEndpoint::class,
-            'identifier' => IdentifierEndpoint::class,
-            'status' => StatusEndpoint::class,
-        ];
-        foreach ($endpoints as $name => $class) {
-            $container->add($name, $class)
-                ->addArgument('rest')
-                ->addArgument(new LiteralArgument($this->getConfigValue('apiBaseUrl')));
-        }
-
-        // Add file download service, without a base class.
-        $container->add('file', Download::class)
-            ->addArgument('rest');
-
-        // Add token endpoint with configuration.
-        $container->add('token_config', new LiteralArgument($this->extractConfigValues([
-            'username',
-            'password',
-            'client',
-        ])));
-        $container->add('auth', TokenEndpoint::class)
-            ->addArgument('rest')
-            ->addArguments([
-                new LiteralArgument($this->getConfigValue('apiBaseUrl')),
-                'token_config',
-            ]);
-
-        // Keep a reference to the container.
-        $this->container = $container;
-    }
-
-    public function setToken(Token $token): self
+    public function setToken(Token $token): ApiClient
```

Comment:
This may seem a little controversial as it breaks the clean architecture, but on the other hand - it keeps things really simple. The token is passed to the factory class.